runtime/datetime.c

```diff
@@ -1258,6 +1260,64 @@ timeConvertToUTC(const struct syslogTime *const __restrict__ local,
 	timeval2syslogTime(&tp, utc, 1);
 }
 
+/**
+ * Format a UNIX timestamp.
+ */
+static int
+formatUnixTimeFromTime_t(time_t unixtime, const char *format, char *pBuf, uint pBufMax) {
+	struct tm lt;
+
+	assert(format != NULL);
+	assert(pBuf != NULL);
+
+	// Let's not depend on debug asserts to save the day in the wild!
+	if (format == NULL) {
+		DBGPRINTF("format is NULL!");
+		return -1;
+	}
+	if (pBuf == NULL) {
+		DBGPRINTF("pBuf is NULL!");
+		return -1;
+	}
+
+	// Convert to struct tm
+	if (gmtime_r(&unixtime, &lt) == NULL) {
+		DBGPRINTF("Unexpected error calling gmtime_r().\n");
+		return -1;
+	}
+
+	// Do our conversions
+	if (strcmp(format, "date-rfc3164") == 0) {
+		if (pBufMax < 16) {
```

Comment:
I would tend to do the next two statements as assert() as well, but am open to discussion. @davidelang what do you think?